services/src/test/scala/cromwell/pipeline/service/ProjectFileServiceTestImpl.scala

```diff
@@ -0,0 +1,60 @@
+package cromwell.pipeline.service
+import cromwell.pipeline.datastorage.dto._
+import cromwell.pipeline.model.wrapper.UserId
+
+import java.nio.file.Path
+import scala.concurrent.Future
+
+class ProjectFileServiceTestImpl(projectFiles: Seq[ProjectFile], testMode: TestMode) extends ProjectFileService {
+
+  override def validateFile(fileContent: ProjectFileContent): Future[Either[ValidationError, Unit]] =
+    testMode match {
+      case WithException(exc: ValidationError) => Future.successful(Left(exc))
+      case WithException(exc)                  => Future.failed(exc)
+      case _                                   => Future.successful(Right(Unit))
```

Comment:
```suggestion
      case _                                   => Future.successful(Right(()))
```
`Unit` here is not the same thing as `()`. `Unit` is a type that has a companion object called `Unit` (the one that you return here) and the instance of `Unit` can be created with `()`. Here it compiles because Scala treats methods that return `Unit` specially, you can write 
```suggestion
      case _                                   => Future.successful(Right("hello"))
```
It will work, be you shouldn't do it :)

Reply:
Shame on me. Fixed